ezomero/ezomero.py

```diff
@@ -66,17 +66,28 @@ def post_dataset(conn, dataset_name, project_id=None, description=None):
     if type(description) is not str and description is not None:
         raise TypeError('Dataset description must be a string')
 
+    project = None
+    current_group = conn.SERVICE_OPTS.getOmeroGroup()
+    if project_id is not None:
+        if type(project_id) is not int:
+            raise TypeError('Project ID must be integer')
+        conn.SERVICE_OPTS.setOmeroGroup('-1')
+        project = conn.getObject('Project', project_id)
+        if project is not None:
+            conn.SERVICE_OPTS.setOmeroGroup(project.getDetails().group.id.val)
```

Comment:
Would it make sense to use `set_group` instead here? It would allow for a "better" failure in case the user is not a member of the group where the project is located. Otherwise, I like this idea enough that I've opened #5 to extend this concept to all functions where relevant!

Reply:
Sounds good. Done.